packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -151,6 +156,39 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
             ;
     }
 
+    renderYAxis(): React.ReactNode {
+        // Y axis with D3
+        const chartHeight = parseInt(this.props.style.height.toString());
+        const margin = { top: 15, right: 0, bottom: 5, left: this.props.style.yAxisWidth };
+        let allMax = 0;
+        this.state.xyData?.datasets?.forEach((dSet: any) => {
+            const rowMax = Math.max(...dSet.data);
+            allMax = Math.max(allMax, rowMax);
+        });
+        const yScale = scaleLinear()
+            .domain([0, Math.max(allMax, 1)])
+            .range([chartHeight - margin.bottom, margin.top]);
+        const yTransform = `translate(${margin.left}, 0)`;
+        // Abbreviate large numbers
+        const scaleYLabel = (d: number) => (
+            d >= 1000000000000 ? d / 1000000000000 + 'G' :
+            d >= 1000000000 ? d / 1000000000  + 'B':
+            d >= 1000000 ? d / 1000000 + 'M' :
+            d >= 1000 ? d / 1000 + 'K':
+            d
```

Comment:
Should support small numbers also? milli micro nano?

Reply:
True! I didn't think of this at all!

Reply:
Do we have a concrete example of when numbers would be very small? The latency scatter plots?

Reply:
By the way,  the TSP currently provides a descriptions of the axis (x or y), with a data type, unit etc. With this, we can decide on how to format the axis (like it's done in Eclipse Trace Compass). However, the Trace Compass server doesn't serialize these parameters at the moment. 

So, the formatting of the y axis, including other case than the one currently provided in this PR, can be done in a separate task, that includes back-end and front-end updates. 

https://github.com/theia-ide/tsp-typescript-client/blob/master/tsp-typescript-client/src/models/xy.ts

Reply:
We can add support for small numbers later if necessary.